app/javascript/packs/colorPicker.jsx

```diff
@@ -0,0 +1,28 @@
+import { h, render } from 'preact';
+import { ColorPicker } from '@crayons';
+
+// Find any color picker inputs on the page and replace them with the Preact enhanced component
+const colorInputs = document.querySelectorAll('[data-color-picker]');
+
+for (const input of colorInputs) {
+  const { labelText } = input.dataset;
+  const inputProps = {};
+
+  // Copy any specific attributes to the new input
+  const { attributes: inputAttributes } = input;
+  for (const attr of inputAttributes) {
+    inputProps[attr.name] = attr.value;
+  }
+
+  render(
+    <ColorPicker
+      id={inputProps.id}
+      defaultValue={input.value}
+      inputProps={inputProps}
+      buttonLabelText={labelText}
+    />,
+    input.parentElement,
+    input,
+  );
+  input.remove();
```

Comment:
There is some content layout shift occurring when the input is replaced by the Preact component.

https://user-images.githubusercontent.com/833231/154353860-6931c51e-e1d5-4e85-906f-7d1c1f1b4556.mp4

Some CSS can probably rework this to avoid the shift. Happy to brainstorm on this.

Reply:
Hmm when I test it, there's no layout shift - this is what I'm seeing (I've watched it at half speed to be double sure!):

https://user-images.githubusercontent.com/20773163/154439240-2db55061-7b0a-4459-98c2-e85cf1ee3f46.mp4

I notice in your video the input is just completely in the wrong place 😄  Did you have the third prop removed from `render()`? I found that although the third prop wasn't causing the input to be removed, it was making sure Preact rendered it in the correct place, which is why I had it in there 🤔

Reply:
Ahh, didn't realize that that was why you had it there. I'll take this for a spin again then.

Reply:
Yeah, this looks great. Consider adding a comment as to why the replace node and remove node are there.